contrib/initramfs/scripts/zfs.in

```diff
@@ -864,8 +864,11 @@ mountroot()
 	pool="$("${ZPOOL}" get name,guid -o name,value -H | \
```

Comment:
Sorry for long response, thank you for tackling this one! 

Looks like we just need to rework this line, so to get something only on guid:

```diff
-pool="$("${ZPOOL}" get name,guid -o name,value -H | \
+pool="$("${ZPOOL}" get guid -o name,value -H | \
```

Could you update the PR? Or I can create the new one.

Reply:
I'm not sure this actually fixes the issue.  Lines 864+865 has zpool create a table of first column name, second column either name or guid and makes awk find the $ZFS_RPOOL value in column 2 and return the name found in column 1, and set this as $pool.

For example:
```
root@server:~# zpool get name,guid -o name,value -H
tank    tank
tank    3542733007923249297
temproot        temproot
temproot        12716802561341322650
```

So, $ZFS_RPOOL was either "3542733007923249297" or "tank", it would return "tank", therefore, it's always safe running this because GUID return name and name returns name.

The problem arises when ZFS_BOOTFS is rebuilt in #867.  I just did a search/replace instead of a rebuild.  If GUID not found, no foul, if found, fixed.  For completeness, I used the same technique for ZFS_RPOOL.

Reply:
@ghfields lines 864-871 were added only for GUID case, so it will be ok to do anything in 867-868 only for GUIDs. Please see original patch https://github.com/zfsonlinux/zfs/commit/e871a8f0584a3259c342f8e3e817159600302832#diff-bbe18a62c620b69e6999ed859dc87d0c

Reply:
If someone puts OS root in zfs root dataset and refers to it by GUID on the kernel commandline, without correcting Line 868, they will still run into the same issue of it trying to mount "rpool/rpool" instead of just "rpool"

Reply:
Let's apply mine fix for line 868 into yours and it's ready to go, so it would be the best one, I'll test it on my laptop.

Reply:
I think you are saying to use the my search/replace for ZFS_BOOTFS, and leave the ZFS_RPOOL as is. I've done this change and retested. (also changed the 'zpool' to ${ZPOOL} within the sed command.